machine/machine_core/ssh_connection.py

```diff
@@ -30,6 +30,16 @@
 from . import timeout as timeoutlib
 
 
+def write_all(fd, data):
+    while len(data) > 0:
+        try:
+            written = os.write(fd, data)
+            data = data[written:]
+        except BlockingIOError:
+            time.sleep(0.1)
```

Comment:
```suggestion
        select.select([], [fd], [])
        written = os.write(fd, data)
        data = data[written:]
```

Reply:
Thanks! Sent to PR #2112 . If that works (hopefully should), let's land that one. Otherwise this will be ready to land.